packages/csvviewer-extension/src/searchprovider.ts

```diff
@@ -2,14 +2,18 @@
 // Distributed under the terms of the Modified BSD License.
 import { ISearchProvider, ISearchMatch } from '@jupyterlab/documentsearch';
 import { CSVViewer } from '@jupyterlab/csvviewer';
-import { IDocumentWidget, DocumentWidget } from '@jupyterlab/docregistry';
+import { DocumentWidget } from '@jupyterlab/docregistry';
 import { Signal, ISignal } from '@phosphor/signaling';
+import { Widget } from '@phosphor/widgets';
 
-export class CSVSearchProvider implements ISearchProvider {
+// The type for which canSearchFor returns true
+export type CSVDocumentWidget = DocumentWidget & { content: CSVViewer };
```

Comment:
Was not `IDocumentWidget<CSVViewer>` sufficient in this case?

Reply:
It would be sufficient. I figured that the form I currently have more accurately reflects the type that `canSearchOn` actually checks for:

https://github.com/jupyterlab/jupyterlab/blob/05c46e3f64338373cb68df2e6de831a8a91c9702/packages/csvviewer-extension/src/searchprovider.ts#L16-L22

Reply:
I think the `IDocumentWidget<CSVViewer>` would make it clearer to the user what this type actually represents. It also ties the types more together in case the interface `IDocumentWidget` is updated in the future. Also, I noticed there is a typo in the comment: s/CMSearchProvider/CSVSearchProvider/ 😄

Reply:
done